Flaky test: test_deep_search_no_results fails due to test isolation issue

## Description

The `test_deep_search_no_results` test in `tests/src/e2e/tools/test_deep_search.py` is flaky and fails intermittently due to a test isolation issue.

## Failure Details

**Test:** `tests/src/e2e/tools/test_deep_search.py::test_deep_search_no_results`

**Error:**
```
AssertionError: Should have no helper matches
assert 1 == 0
 +  where 1 = len([{'entity_id': 'input_text.concurrent_test_3', 'helper_type': 'input_text', 'name': 'Concurrent Test 3', 'score': 75, 'match_in_name': False, 'match_in_config': True, 'config': {'id': 'concurrent_test_3', 'name': 'Concurrent Test 3', 'mode': 'text', 'max': 100, 'min': 0}}])
```

**Root Cause:**
The test searches for a nonsense term and expects zero results, but it's finding a leftover helper entity (`input_text.concurrent_test_3`) from a previous test. This indicates insufficient test cleanup or isolation between test runs.

## Observed Behavior

- Test expects: 0 helper matches when searching for nonsense
- Test gets: 1 helper match (`input_text.concurrent_test_3`)
- The helper appears to be leftover from concurrent test execution

## Impact

- **Severity:** Low (flaky test, not a production bug)
- **Frequency:** Intermittent
- **Workaround:** Retry CI or use admin override for merge

## Recent Occurrences

- PR #75 (Dashboard Management): https://github.com/homeassistant-ai/ha-mcp/actions/runs/19432194135/job/55593672066
- Run date: 2025-11-17

## Recommended Fix

**Option 1: Improve test cleanup**
- Ensure all helper entities created during tests are properly cleaned up in teardown
- Check for leftover entities from `test_helper_bulk_operations` and similar tests

**Option 2: Improve test isolation**
- Use unique naming schemes for test helpers to avoid conflicts
- Consider test-specific prefixes or UUIDs

**Option 3: Make assertion more robust**
- Instead of asserting zero results, filter out known test helpers
- Or skip this specific assertion if test cleanup is inherently racy

## Test Location

File: `tests/src/e2e/tools/test_deep_search.py:336`

```python
async def test_deep_search_no_results(mcp_client):
    """Test that searching for nonsense returns no results."""
    logger.info("🔍 Testing deep search with no matches")
    mcp = MCPAssertions(mcp_client)

    data = await mcp.call_tool_success("ha_deep_search", {"query": "xyznonexistent123"})

    # ... assertions ...
    helpers = data.get("helpers", [])
    assert len(helpers) == 0, "Should have no helper matches"  # ← Fails here
```

## Context

This failure occurred during PR #75 (Dashboard Management feature) but is unrelated to the dashboard code. The test was bypassed with admin override to merge the dashboard feature.

--- a/tests/src/e2e/tools/test_deep_search.py
+++ b/tests/src/e2e/tools/test_deep_search.py
@@ -327,13 +327,15 @@ async def test_deep_search_no_results(mcp_client):
 
     # Verify we get empty results
     # Filter out any test entities that may not have been cleaned up from parallel tests
-    # Common test entity prefixes: deep_search, concurrent_test, test_, e2e_
-    test_prefixes = ["deep_search", "concurrent_test", "test_", "e2e_", "bulk_"]
+    # Common test entity prefixes: deep_search, concurrent_test, test_, e2e_, bulk_
+    test_prefixes = ("deep_search", "concurrent_test", "test_", "e2e_", "bulk_")
 
     def is_test_entity(entity_id: str) -> bool:
         """Check if entity_id appears to be from a test."""
-        entity_lower = entity_id.lower()
-        return any(prefix in entity_lower for prefix in test_prefixes)
+        # Extract object_id (part after domain) to avoid false positives
+        # e.g., "input_text.concurrent_test_3" -> "concurrent_test_3"
+        object_id = entity_id.lower().split('.')[-1]
+        return object_id.startswith(test_prefixes)
 
     automations = [a for a in data.get("automations", []) if not is_test_entity(a.get("entity_id", ""))]
     scripts = [s for s in data.get("scripts", []) if not is_test_entity(s.get("entity_id", ""))]

